scss/_base_typography-definitions.scss

```diff
@@ -204,10 +206,22 @@
   }
 
   %small-caps-text {
-    @extend %common-default-text-properties;
+    @extend %default-text;
     font-variant-caps: all-small-caps;
     letter-spacing: 0.05em;
     margin-bottom: map-get($sp-after, default-text) - map-get($nudges, p);
+
+    &.u-no-margin--bottom {
+      @extend %u-no-margin--bottom--small;
+    }
+
+    // deprecated: the use of .u-align-text--x-small-to-default on small caps text is deprecated
+    // previously small caps was implemented as x-small text variant requiring .u-align-text--x-small-to-default compensations
+    // but now it's implemented with default text size not requiring any compensations,
+    // so we need to reset the padding-top to the default value
+    &.u-align-text--x-small-to-default {
+      padding-top: map-get($nudges, p) + map-get($browser-rounding-compensations, p);
+    }
```

Comment:
I'm confused here, if `.u-align-text--x-small-to-default` is deprecated, why do we need styles for it?

Reply:
Ah ok no yeah I get it! I think it's sensible the way you did it. Do we have any strategy for handling backward compatibility like this? I mean do we drop these kinds of rules in the next major version or does it stay here "forever"?

Reply:
We drop them in major version, because that's when we can break things and write a proper migration guide.

Reply:
That's why I try to mark them with `// deprecated` comment in SCSS so it is easier to find. Technically there is also a `@deprecated` mixin, but TBH, I haven't seen a great value in it, and it complicates the code by nesting everything one more level.

Reply:
Please note the `.u-align-text--x-small-to-default` utility might still be useful because we are still using the small size for table headers.

Reply:
Yes, I'm only deprecating it when used on `p-text--x-small-capitalised` (and its new name `p-text--small-caps`), it is still valid utility to use with other x-small text.